orc8r/cloud/go/services/bootstrapper/servicers/registration/registration_test.go

```diff
@@ -0,0 +1,177 @@
+	rpcErr := status.Error(codes.NotFound, "errMessage")
+
+	registrationServicer := setupMockRegistrationServicer(t)
+	registrationServicer.GetControlProxy = func(networkID string) (string, error) {
+		return "", rpcErr
+	}
+
+	res, err := registrationServicer.Register(nil, registerRequest)
+	assert.NoError(t, err)
+	expectedRes := &protos.RegisterResponse{
+		Response: &protos.RegisterResponse_Error{
+			Error: fmt.Sprintf("error getting control proxy: %v", rpcErr),
+		},
+	}
+	assert.Equal(t, expectedRes, res)
+}
+
+func TestGetControlProxy_NoNetworkID(t *testing.T) {
+	setupAddNetworksToTenantsService(t)
+
+	res, err := registration.GetControlProxy(networkID)
+	assert.Equal(t, status.Errorf(codes.NotFound, "tenantID for current NetworkID %v not found", networkID), err)
+	assert.Equal(t, "", res)
+}
+
+func TestGetControlProxy_NoControlProxy(t *testing.T) {
+	setupAddNetworksToTenantsService(t)
+
+	networkIDTenant := &protos.Tenant{
+		Name:     "tenant",
+		Networks: []string{networkID},
+	}
+	addTenant(t, networkIDTenant)
+
+	res, err := registration.GetControlProxy(networkID)
+	assert.Equal(t, "Not found", err.Error())
+	assert.Equal(t, "", res)
+}
+
+func TestGetControlProxy(t *testing.T) {
+	setupAddNetworksToTenantsService(t)
+
+	networkIDTenant := &protos.Tenant{
+		Name:     "tenant",
+		Networks: []string{networkID},
+	}
+	id := addTenant(t, networkIDTenant)
+	err := tenants.CreateOrUpdateControlProxy(context.Background(), protos.CreateOrUpdateControlProxyRequest{
+		Id:           id,
+		ControlProxy: controlProxy,
+	})
+	assert.NoError(t, err)
+
+	res, err := registration.GetControlProxy(networkID)
+	assert.NoError(t, err)
+	assert.Equal(t, controlProxy, res)
+}
+
+func newMockRegistrationService() *registration.RegistrationService {
+	registrationService := &registration.RegistrationService{
+		GetGatewayDeviceInfo: func(ctx context.Context, token string) (*protos.GatewayDeviceInfo, error) {
+			return gatewayDeviceInfo, nil
+		},
+		RegisterDevice: func(deviceInfo protos.GatewayDeviceInfo, hwid *protos.AccessGatewayID, challengeKey *protos.ChallengeKey) error {
+			return nil
+		},
+		GetControlProxy: func(networkID string) (string, error) {
+			return controlProxy, nil
+		},
+	}
+
+	return registrationService
+}
+
+func setupAddNetworksToTenantsService(t *testing.T) {
+	var (
+		tenant1 = &protos.Tenant{
+			Name:     "tenant",
+			Networks: []string{"network1", "network2"},
+		}
+		tenant2 = &protos.Tenant{
+			Name:     "tenant",
+			Networks: []string{"network3", "network4"},
+		}
+	)
+	tenantsTestInit.StartTestService(t)
+
+	addTenant(t, tenant1)
+	addTenant(t, tenant2)
+}
+
+func addTenant(t *testing.T, tenant *protos.Tenant) int64 {
+	ctx := context.Background()
+
+	tenantRes, err := tenants.CreateTenant(ctx, nextTenantID, tenant)
+	assert.NoError(t, err)
+	assert.Equal(t, tenant, tenantRes)
+
+	curTenantID := nextTenantID
+	nextTenantID = nextTenantID + 1
+	return curTenantID
```

Comment:
I feel like there's definitely a way to have this in a single line -- input requested ty :)